macros/StatusVmon904HV.py

```diff
@@ -0,0 +1,1339 @@
+#!/usr/bin/env python
+
+import cx_Oracle
+import ROOT
+import os
+import time
+from datetime import datetime
+from operator import itemgetter
+from array import array
+
+#in the DB the DeltaV between pins are saved, not the V from ground
+
+#time interval considered to extract the data
+#sta_period = "'2018-07-17 08:00:01'"
+#end_period = "'2018-07-17 14:30:01'"
+
+#to control the status there are data in the period 
+#sta_period = "'2018-12-06 22:00:01'"
+#end_period = "'2019-03-15 08:00:01'"
+
+#test with board 4 channel 2 : 2-2-Top
+
+sta_period = raw_input("Insert UTC start time in format YYYY-MM-DD HH:mm:ss\n")
+type(sta_period)
+end_period = raw_input("Insert UTC end time in format YYYY-MM-DD HH:mm:ss\n")
+type(end_period)
+
+#to remove ' and space
+start=sta_period.replace(" ", "_")
+end=end_period.replace(" ", "_")
+#start = start[1:-1]
+#end = end[1:-1]
+start=start.replace(":", "-")
+end=end.replace(":", "-")
+#print(start)
+#print(end)
+
+#add ' at beginning and end to have the date in the format for the query
+sta_period = "'" + sta_period + "'"
+end_period = "'" + end_period + "'"
+#print(sta_period)
+#print(end_period)
+
+# I also include some root histogram in case some plot are needed..
+fileName = "QC8_HV_monitor_UTC_start_"+start+"_end_"+end+".root"
+f1=ROOT.TFile( fileName,"RECREATE")
+
+#divide the monitoring period in three periods 
+#the limits between period 1 and 2 is defined by the first  date in which we changed the mapping of HV: 03-Apr-2019 13.10
+#the limits between period 2 and 3 is defined by the second date in which we changed the mapping of HV: 15-Apr-2019 08:15
+
+#in the second date only the LV mapping is changed, but the datapoints change. So we define three periods even if we have
+#only two mappings
+
+mappingChangeDate = []
+firstMappingChange 	= datetime( 2019, 04, 03, 13, 10, 00 )
+secondMappingChange 	= datetime( 2019, 04, 15,  8, 15, 00 ) 
+mappingChangeDate.append( firstMappingChange )
+mappingChangeDate.append( secondMappingChange )
+
+startDate = datetime(int(start[:4]), int(start[5:7]), int(start[8:10]), int(start[11:13]), int(start[14:16]), int(start[17:]) )
+endDate	  = datetime(int(end[:4]), int(end[5:7]), int(end[8:10]), int(end[11:13]), int(end[14:16]), int(end[17:]) )
+
+periodBool = []
+numberOfMaps = 3 #date 10-May-2019
+
+#create a vector of numbers full of zero: numbero of zero is equal to the number of maps
+for mapIdx in range(numberOfMaps):
+	periodBool.append(0)
+
+startIdx = -1
+endIdx = -1
+#find the index for mapping start and end
+
+if startDate < mappingChangeDate[0]: 
+	startIdx = 0
+elif (startDate > mappingChangeDate[0] and startDate < mappingChangeDate[1]):
+	startIdx = 1
+elif startDate > mappingChangeDate[1]:
+	startIdx = 2
+#print "startIdx:", startIdx
+
+if endDate < mappingChangeDate[0]: 
+	endIdx = 0
+elif endDate > mappingChangeDate[0] and endDate < mappingChangeDate[1]:
+	endIdx = 1
+elif endDate > mappingChangeDate[1]:
+	endIdx = 2
+#print "endIdx:", endIdx
```

Comment:
Lines 75-89 could be done with a function, e.g.

```python
def determineDateIdx(date, mappingChangeDate):
    if date < mappingChangeDate[0]: 
    	return 0
    elif (date > mappingChangeDate[0] and startDate < mappingChangeDate[1]):
	return 1
    elif date > mappingChangeDate[1]:
	return 2
```

Then:

```python
startIdx = determineDateIdx(startDate, mappingChangeDate)
endIdx = determineDateIdx(startDate, mappingChangeDate)
```

If this is the only place that `mappingChangeDate` is used then it could also be defined solely in the above function.